ocs_ci/ocs/jenkins.py

```diff
@@ -0,0 +1,265 @@
+
+    def get_jenkins_deploy_pods(self):
+        """
+        Get all jenkins deploy pods
+
+        Returns:
+            List: jenkins deploy pod objects list
+
+        """
+        return [
+            get_pod_obj(
+                pod
+            ) for pod in get_pod_name_by_pattern('deploy', namespace=constants.JENKINS_NAMESPACE)
+        ]
+
+    def get_builds_obj(self):
+        """
+        Get all jenkins builds
+
+        Returns:
+            List: jenkins deploy pod objects list
+
+        """
+        build_obj_list = []
+        build_list = self.get_build_name_by_pattern(
+            pattern=constants.JENKINS_BUILD, namespace=constants.JENKINS_NAMESPACE
+        )
+        for build_name in build_list:
+            ocp_obj = OCP(api_version='v1', kind='Build', namespace=constants.JENKINS_NAMESPACE)
+            ocp_dict = ocp_obj.get(resource_name=build_name)
+            build_obj_list.append(OCS(**ocp_dict))
+        return build_obj_list
+
+    def create_jenkins_pvc(self):
+        """
+        create jenkins pvc
+
+        """
+        pvc_obj = create_pvc(
+            pvc_name='dependencies', size='10Gi', sc_name='ocs-storagecluster-ceph-rbd',
+            namespace=constants.JENKINS_NAMESPACE
+        )
+        return pvc_obj
+
+    def create_app_jenkins(self):
+        """
+        create application jenkins
+
+        """
+        cmd = 'new-app --name=jenkins-ocs-rbd --template=jenkins-persistent-ocs'
+        self.ocp.exec_oc_cmd(command=cmd, out_yaml_format=False)
+
+    def start_build(self):
+        """
+        Start build on jenkins
+
+        """
+        log.info("Start build on jenkins")
+        cmd = f"start-build {constants.JENKINS_BUILD}"
+        self.ocp.exec_oc_cmd(command=cmd, out_yaml_format=False)
+
+    def setup_jenkins(self):
+        """
+        Setup jenkins
+
+        """
+        self.create_ocs_jenkins_template()
+        self.create_jenkins_pvc()
+        self.create_app_jenkins()
+        self.setup_jenkins_build_config()
+        self.wait_for_jenkins_deploy_status(status=constants.STATUS_COMPLETED)
+
+    def get_build_name_by_pattern(
+        self,
+        pattern='client',
+        namespace=None,
+        filter=None,
+    ):
+        """
+        Get build name by pattern
+
+        Returns:
+            list: build name
+
+        """
+        ocp_obj = OCP(kind='Build', namespace=namespace)
+        build_names = ocp_obj.exec_oc_cmd('get build -o name', out_yaml_format=False)
+        build_names = build_names.split('\n')
+        build_list = []
+        for name in build_names:
+            if filter is not None and re.search(filter, name):
+                log.info(f'build name filtered {name}')
+            elif re.search(pattern, name):
+                (_, name) = name.split('/')
+                log.info(f'pod name match found appending {name}')
+                build_list.append(name)
+        return build_list
+
+    def get_jenkins_nodes(self):
```

Comment:
We have a function to get the pod node which does the below steps.
get_pod_node in pod.py.
Please check